torch/lib/TH/THSize.h

```diff
@@ -0,0 +1,13 @@
+#ifndef TH_SIZE_INC
+#define TH_SIZE_INC
+
+#include "THGeneral.h"
+#include <stddef.h>
+
+// THTensor functions that would work on a THSize if we had such a class in C++,
+// i.e. THTensor functions that depend only on the shape of the tensor, not the type.
```

Comment:
I don't think we should be adding additional uses of THLongStorage to represent sizes. It was a mistake to represent sizes that way that only sort-of made sense in the context of Lua.

We already use `long*` pointers in `resizeNd` and `setStorageNd`.

It's also bad because we take the THLongStorage.size -> THTensor.nDimension, but one is an `int` and the other is a `long`. This requires either `-fpermissive` everywhere or extra casts.